src/Persistent_cohomology/include/gudhi/Persistent_cohomology/Field_Zp.h

```diff
@@ -34,15 +35,34 @@ class Field_Zp {
 
   void init(int charac) {
     assert(charac > 0);  // division by zero + non negative values
+
     Prime = charac;
+
+    // Check that the provided prime is less than the maximum allowed as int and calculation below : 46337 ; i.e (max_prime-1)**2 <= INT_MAX
```

Comment:
I think we actually need `(max_prime-1)*max_prime<=INT_MAX` for plus_times_equal.